design_docs/analysis_task_template.md

```diff
@@ -0,0 +1,594 @@
+        across various versions of MPAS components where stream names have
+        changed.
+
+    variableMap : dict, optional
+        A dictionary of MPAS component variable names that map to their
+        MPAS-Analysis counterparts. These are used to maintain compatibility
+        across various versions of MPAS components where variable names have
+        changed.
+
+    fieldName : {'sst', 'mld', 'sss'}
+        The name of the field to be analyzed
+
+    kwargs : dict
+        A dictionary of remaining keyword arguments (ignored in this function)
+
+    Raises
+    ------
+    ValueError
+        If fieldName was not supplied
+    OSError
+        If files are not present
+
+    Authors
+    -------
+    Xylar Asay-Davis
+
+    Last Modified
+    -------------
+    03/08/2017
+    """
+
+    possibleFieldNames = ['sst', 'mld', 'sss']
+    if fieldName not in possibleFieldNames:
+        raise ValueError('fieldName must be one of {}.'.format(
+            possibleFieldNames))
+
+    section = 'climatology'
+    startDate = '{:04d}-01-01_00:00:00'.format(
+        config.getint(section, 'startYear'))
+    if not config.has_option(section, 'startDate'):
+        config.set(section, 'startDate', startDate)
+    endDate = '{:04d}-12-31_23:59:59'.format(
+        config.getint(section, 'endYear'))
+    if not config.has_option(section, 'endDate'):
+        config.set(section, 'endDate', endDate)
+
+    if config.get('runs', 'preprocessedReferenceRunName') != 'None':
+            check_path_exists(config.get('oceanPreprocessedReference',
+                                         'baseDirectory'))
+
+    make_directories(build_config_full_path(config, 'output',
+                                            'plotsSubdirectory'))
+
+    return  # }}}
+```
+This code has been taken out of `run_analysis.py`, simplifying and clarifying
+the code.
+
+<h4> Implementation: Analysis Continues even when Analysis Task Fails <br>
+Date last modified: 2017/03/08 <br>
+Contributors: Xylar Asay-Davis
+</h4>
+
+Calls to `check` and `run` functions in `run_analysis.py` are inside of 
+`try/except` blocks, which catch the exceptions and print the stack trace 
+but don't cause the code to exit.
+```python
+try:
+    analysisModule.check(config, **kwargs)
+    add = True
+except:
+    traceback.print_exc(file=sys.stdout)
+    print "ERROR: analysis module {} failed during check and " \
+        "will not be run".format(taskName)
+
+...
+
+try:
+    analysisModule.run(config, **kwargs)
+except:
+    traceback.print_exc(file=sys.stdout)
+    taskName = analysisModule.get_task_name(**kwargs)
+    print "ERROR: analysis module {} failed during run".format(
+        taskName)
+
+```
+
+<h4> Implementation: List of Tasks to Perform<br>
+Date last modified: 2017/03/08 <br>
+Contributors: Xylar Asay-Davis
+</h4>
+
+The tasks are imported and added to an anaysis list as follows:
+```python
+analyses = []
+
+# Ocean Analyses
+from mpas_analysis.ocean import ohc_timeseries
```

Comment:
The point is that we want to make it clear that there is equivalence between that set of commands and the functional equivalent.  All we need is a comment in the most obvious and/or first place this type of syntax is used.